### PR TITLE
Unify rollback & deployment, simplify deployment workflow

### PR DESCRIPTION
**Pull request by AetherUnbound**
## Fixes

Related to https://github.com/WordPress/openverse/issues/384

## Description

This PR incorporates the changes proposed in https://github.com/WordPress/openverse-infrastructure/pull/378 and unifies the rollback & deployment into a single action. It also simplifies the deployment workflow itself by templating which action to use based on the inputs. Lastly, it incorporates a fix that @dhruvkb identified in https://github.com/WordPress/openverse/pull/442 regarding the `image-tag` input to the CI/CD workflow.

## Testing Instructions

I _think_ we'll need to merge this to test it, but I'll see if I can launch the workflow from this branch as well prior to merging.

## Checklist

- [x] My pull request has a descriptive title (not a vague title like
      `Update index.md`).
- [x] My pull request targets the _default_ branch of the repository (`main`) or
      a parent feature branch.
- [x] My commit messages follow [best practices][best_practices].
- [x] My code follows the established code style of the repository.
- [ ] I added or updated tests for the changes I made (if applicable).
- [x] I added or updated documentation (if applicable).
- [x] I tried running the project locally and verified that there are no visible
      errors.

[best_practices]:
  https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin

<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>

### CODE REVIEW

File: .github/actions/staging-api-deploy/action.yml [hunk 1]
```diff
@@ -66,6 +66,62 @@ runs:
...
+            }
+            exists = versions.some((v) => v.metadata.container.tags.includes('${{ inputs.tag }}'))
+          }
+          if (!exists) {
+            throw new Error(
+              '`${{ inputs.tag }}` does not appear to be a valid tag for the ghcr.io/wordpress/openverse-api-nginx:latest image.'
+            )
+          }
+
+    - name: Validate `tag` input for django
```
Comment (dhruvkb) [inline .github/actions/staging-api-deploy/action.yml:97]:
```suggestion
    - name: Validate `tag` input for Django
```

File: .github/actions/staging-api-deploy/action.yml [hunk 2]
```diff
@@ -90,7 +146,7 @@ runs:
       with:
         task-definition: task-definition.json
         container-name: nginx
-        image: "ghcr.io/wordpress/openverse-api_nginx:${{ inputs.tag }}"
+        image: "ghcr.io/wordpress/openverse-api-nginx:${{ inputs.tag }}"
```
Comment (dhruvkb) [inline .github/actions/staging-api-deploy/action.yml:149]: This was correct earlier. The image on GHCR: https://github.com/WordPress/openverse/pkgs/container/openverse-api_nginx

File: .github/actions/staging-nuxt-deploy/action.yml [hunk 3]
```diff
@@ -66,6 +66,34 @@ runs:
         SLACK_WEBHOOK_URL: ${{ inputs.slack-webhook }}
         SLACK_WEBHOOK_TYPE: INCOMING_WEBHOOK
 
+    - name: Validate `tag` input for nuxt
```
Comment (dhruvkb) [inline .github/actions/staging-nuxt-deploy/action.yml:69]:
```suggestion
    - name: Validate `tag` input for Nuxt
```

**Review by dhruvkb — CHANGES_REQUESTED** [review]
I like the changes but I have one note about the `-`/`_` in the API-NGINX image (and some nits about casing in the step names).

File: .github/actions/staging-api-deploy/action.yml [hunk 4]
```diff
@@ -66,6 +66,62 @@ runs:
...
+                per_page: 100,
+              })
+            if (!versions.length) {
+              break
+            }
+            exists = versions.some((v) => v.metadata.container.tags.includes('${{ inputs.tag }}'))
+          }
+          if (!exists) {
+            throw new Error(
+              '`${{ inputs.tag }}` does not appear to be a valid tag for the ghcr.io/wordpress/openverse-api-nginx:latest image.'
```
Comment (obulat) [inline .github/actions/staging-api-deploy/action.yml:93]: Shouldn't the quotes change places: 
```suggestion
              `'${{ inputs.tag }}' does not appear to be a valid tag for the ghcr.io/wordpress/openverse-api-nginx:latest image.`
```

**Comment by AetherUnbound** [comment]
Thanks for the eyes on this @dhruvkb and @obulat! All of the stuff y'all commented on are from auto-generated files, so I'll take your suggestions upstream to https://github.com/WordPress/openverse-infrastructure/pull/378 and apply them here as well 🙂

File: .github/actions/staging-api-deploy/action.yml [hunk 5]
```diff
@@ -66,6 +66,62 @@ runs:
...
+      with:
+        script: |
+          let exists = undefined,
+            page = 0
+          while (!exists) {
+            page += 1
+            const { data: versions } =
+              await github.rest.packages.getAllPackageVersionsForPackageOwnedByOrg({
+                package_type: 'container',
+                package_name: 'openverse-api-nginx',
```
[the remark this answers is not in the document]
Comment (AetherUnbound) [reply .github/actions/staging-api-deploy/action.yml:80]: Oh I didn't realize! Where/when was this changed? We'll have to update the infrastructure repo with this new image.

[the remark this answers is not in the document]
Comment (AetherUnbound) [reply .github/actions/staging-api-deploy/action.yml:80]: Ah, found it here: https://github.com/WordPress/openverse/commit/9d4c9094d5f079a49cf78b9ecc646ba49c0e2654

File: .github/actions/staging-api-deploy/action.yml (hunk 4 above)
[the remark this answers is not in the document]
Comment (AetherUnbound) [reply .github/actions/staging-api-deploy/action.yml]: Gah, good catch!

File: .github/actions/staging-api-deploy/action.yml [hunk 6]
```diff
@@ -66,6 +66,62 @@ runs:
         SLACK_WEBHOOK_URL: ${{ inputs.slack-webhook }}
         SLACK_WEBHOOK_TYPE: INCOMING_WEBHOOK
 
+    - name: Validate `tag` input for nginx
```
[the remark this answers is not in the document]
Comment (AetherUnbound) [reply .github/actions/staging-api-deploy/action.yml]: Due to this being automatically generated, closest I could get was `Nginx` 😅 hope that's okay!

**Review by zackkrida — APPROVED** [review]
👍 I approved the PR which generates this file, so approving here as well.

**Review by dhruvkb — APPROVED** [review]
LGTM! Thanks for compensating for my ignorance and applying the changes at the right place.

**Pull request merged**